### PR TITLE
ceph.in: allow 'flags' to not be present in cmddescs

### PR DESCRIPTION
**Pull request by dmick**
the 'flags' key in the command description dict may not be
present on older monitors.  Check for existence before
using it.

Introduced in 61d63433dfdcc08ecc21ba4f645794da5e01823c

Fixes: http://tracker.ceph.com/issues/18297
Signed-off-by: Dan Mick <dan.mick@redhat.com>

### CODE REVIEW

File: src/ceph.in [hunk 1]
```diff
@@ -342,7 +342,9 @@ def format_help(cmddict, partial=None):
 
         if not cmd['help']:
             continue
-        if cmd['flags'] is not None and ((cmd['flags'] & (FLAG_OBSOLETE | FLAG_DEPRECATED)) != 0):
+        flags = cmd.get('flags')
+        if (flags is not None and
```
Comment (alfredodeza) [inline src/ceph.in]: this explicit check for `flags is not None` is clearly fixing the issue, but this is extremely hard to parse due to the uncommon usage of Python's bitwise operators.

Is it really that necessary to use logical bitwise operators to check if a flag is obsolete or deprecated?

Comment (liewegas) [inline src/ceph.in]: I don't think it can be changed without chagning hte json schema...

Comment (alfredodeza) [inline src/ceph.in]: I mean, it is not necessary to use `&`, `|`, or `<<` here. Regular Python operators would do just fine.

Comment (dmick) [inline src/ceph.in]: just saw this.  It seems like idiomatic usage to me, but then I'm a C programmer first

**Comment by batrick** [comment]
I was on PTO when you requested this review. It looks good. Thanks for fixing this @dmick.
